proposals/0071-Entity/06-querying.md

```diff
@@ -0,0 +1,555 @@
+# Count requests by pod status
+count by (k8s.pod.status.phase) (rate(http_requests_total[5m]))
+```
+
+### Aggregation Semantics
+
+Aggregation happens **after** enrichment:
+
+```
+1. Select series matching the selector
+2. Enrich each series with entity labels
+3. Group by the specified labels (which may include entity labels)
+4. Apply aggregation function
+```
+
+**Example:**
+
+```promql
+sum by (k8s.node.name) (container_cpu_usage_seconds_total)
+```
+
+```
+Step 1 - Select series:
+  container_cpu{pod_uid="a", node_uid="n1"} 10
+  container_cpu{pod_uid="b", node_uid="n1"} 20
+  container_cpu{pod_uid="c", node_uid="n2"} 30
+
+Step 2 - Enrich with entity labels:
+  container_cpu{..., k8s.node.name="worker-1"} 10
+  container_cpu{..., k8s.node.name="worker-1"} 20
+  container_cpu{..., k8s.node.name="worker-2"} 30
+
+Step 3 - Group by k8s.node.name:
+  Group "worker-1": [10, 20]
+  Group "worker-2": [30]
+
+Step 4 - Sum:
+  {k8s.node.name="worker-1"} 30
+  {k8s.node.name="worker-2"} 30
+```
+
+---
+
+## Range Queries and Temporal Semantics
+
+### The Challenge
+
+Descriptive labels can change over time. When querying a range, which label values should be used?
+
+**Example scenario:**
+- Pod `abc-123` runs on `worker-1` from T0 to T5
+- Pod migrates to `worker-2` at T5
+- Query: `container_cpu_usage_seconds_total{k8s.pod.uid="abc-123"}[10m]`
+
+### Solution: Point-in-Time Label Resolution
+
+Each sample is enriched with the descriptive labels **that were valid at that sample's timestamp**.
+
+```promql
+container_cpu_usage_seconds_total{k8s.pod.uid="abc-123"}[10m]
+```
+
+**Returns:**
+```
+# Samples before migration (T0-T4) have worker-1
+container_cpu{k8s.pod.uid="abc-123", k8s.node.name="worker-1"} 100 @T0
+container_cpu{k8s.pod.uid="abc-123", k8s.node.name="worker-1"} 110 @T1
+container_cpu{k8s.pod.uid="abc-123", k8s.node.name="worker-1"} 120 @T2
+container_cpu{k8s.pod.uid="abc-123", k8s.node.name="worker-1"} 130 @T3
+container_cpu{k8s.pod.uid="abc-123", k8s.node.name="worker-1"} 140 @T4
+
+# Samples after migration (T5+) have worker-2
+container_cpu{k8s.pod.uid="abc-123", k8s.node.name="worker-2"} 150 @T5
+container_cpu{k8s.pod.uid="abc-123", k8s.node.name="worker-2"} 160 @T6
+...
+```
+
+### Implications for Range Functions
+
+Functions like `rate()` operate on the raw sample values, but the returned instant vector has enriched labels:
+
+```promql
+rate(container_cpu_usage_seconds_total{k8s.pod.uid="abc-123"}[5m])
+```
+
+For rate calculation:
+- Uses sample values regardless of label changes
+- The result is enriched with labels **at the evaluation timestamp**
+
+### Series Identity Across Label Changes
+
+**Important:** Descriptive label changes do NOT create new series. The series identity is defined by:
+- Metric name
+- Original metric labels
+- Entity identifying labels (correlation keys)
+
+Descriptive labels are metadata that "rides along" with samples, not part of series identity.
+
+---
+
+## The Entity Type Filter Operator
```

Comment:
Ooooof, a whole new mini language for just one thing.

Is there really no ways this can be modeled with existing label matchers? Put the entity type into a label `entity_type`? Or maybe `__entity_type__` to not make it collide with a normal label that happens to have that name…